src/providers/ec2/ec2.go

```diff
@@ -101,3 +145,45 @@ func (p provider) ShouldRetry() bool {
 func (p *provider) BackoffDuration() time.Duration {
 	return util.ExpBackoff(&p.backoff, maxBackoff)
 }
+
+func fetchBody(client *http.Client, url string, acceptedStatuses ...int) ([]byte, int, error) {
```

Comment:
if we're going to have a generic http body fetcher with enumerated acceptable status codes, it should at least be done such that the callers can be simplified to just handle errors or use the returned body, don't you think? 

I also feel like the fetcher should be doing the logging of what's being fetched, and the callers can dispense with their logging.